packages/react-redux-subspace/src/components/SubspaceProvider.js

```diff
@@ -10,37 +10,50 @@ import React, { Children } from 'react'
 import PropTypes from 'prop-types'
 import { subspace }  from 'redux-subspace'
 
-class SubspaceProvider extends React.PureComponent {
-
-    getChildContext() {
-        const makeSubspaceDecorator = (props) => props.subspaceDecorator || subspace(props.mapState, props.namespace)
+export function createSubspaceProvider(paramStoreKey = 'store') {
+    let storeKey = typeof paramStoreKey === 'object' ? paramStoreKey.storeKey || 'store' : paramStoreKey;
+    let parentStoreKey = typeof paramStoreKey === 'object' ? paramStoreKey.parentStoreKey || 'store' : paramStoreKey;
+
+    class SubspaceProvider extends React.PureComponent {
+        getChildContext() {
+            const makeSubspaceDecorator = (props) => props.subspaceDecorator || subspace(props.mapState, props.namespace)
+
+            if (storeKey !== 'store') {
+                console.log('parentStoreKey: ', parentStoreKey)
+                console.log(this.context[parentStoreKey])
+            }
+            return {
+                [storeKey]: makeSubspaceDecorator(this.props)(this.context[parentStoreKey]),
+                [`${storeKey}Subscription`]: this.context[`${parentStoreKey}Subscription`],
```

Comment:
I'm starting to wonder if it would be easier to just use `react-redux`'s `Provider` to inject our subspaced store back into the context instead of trying to replicate how `Provider` does it.  It might make us more resilient to any changes they may make in the future.  

I won't insist on changing it to do that as part of this PR as I don't anticipate them changing anytime soon (other than the huge context API changes but we'll likely have to overhaul a bunch of stuff when that happens anyway).